src/lib/libwasmfs_fetch.js

```diff
@@ -134,7 +134,7 @@ addToLibrary({
           return failedResponse.status === 404 ? -{{{ cDefs.ENOENT }}} : -{{{ cDefs.EBADF }}};
         }
         var fileInfo = wasmFS$JSMemoryRanges[file];
-        length = Math.min(length, fileInfo.size-offset+1) | 0;
+        length = Math.min(length, fileInfo.size-offset) | 0;
```

Comment:
Do you need the `| 0` here or above?

Reply:
Ah no, I guess they're only needed after divisions